Sources/Xcodeproj/generate().swift

```diff
@@ -70,14 +78,43 @@ public func generate(dstdir: String, projectName: String, srcroot: String, modul
     return xcodeprojPath
 }
 
-private func open(_ path: String..., body: ((String) -> Void) throws -> Void) throws {
+import func libc.fgetc
+import var libc.EOF
+
+private func open(_ path: String..., cacheDir: String, body: ((String) -> Void) throws -> Void) throws {
     let path = Path.join(path)
+    // FIXME: Two files can have same name and different path.
+    // Probably can use MD5 hash of path as file name of the cache.
+    let cachePath = Path.join(cacheDir, path.basename)
     let stream = OutputByteStream()
     try body { line in
         stream <<< line
         stream <<< "\n"
     }
-    try fopen(path, mode: .Write) { fp in
-        try fputs(stream.bytes.bytes, fp)
+    // If file and its cache exists, check if new and old
+    // buffer are equal. Only re-write if they're not same. 
+    if path.isFile && cachePath.isFile {
+        var contents = [UInt8]()
+        try fopen(cachePath, mode: .Read) { fp in
+            while true {
+                let c = fgetc(fp)
+                if c == EOF { break }
+                contents += [UInt8(c)]
+            }
+        }
```

Comment:
This is a very slow way to read the contents of a file, do we not already have something faster? If not, we should factor out something that uses `fread()` or equivalent.